src/main/java/com/serhiiklymchuk/gmailpet/util/mapper/MessageToMessageDtoMapper.java

```diff
@@ -0,0 +1,47 @@
+package com.serhiiklymchuk.gmailpet.util.mapper;
+
+import com.serhiiklymchuk.gmailpet.domain.Message;
+import com.serhiiklymchuk.gmailpet.domain.User;
+import com.serhiiklymchuk.gmailpet.dto.MessageDto;
+import com.serhiiklymchuk.gmailpet.repository.MessageRepository;
+import com.serhiiklymchuk.gmailpet.repository.UserRepository;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Component
+public class MessageToMessageDtoMapper {
+
+    private final UserRepository userRepository;
+
+    public MessageToMessageDtoMapper(UserRepository userRepository, MessageRepository messageRepository) {
+        this.userRepository = userRepository;
+    }
+
+    public List<MessageDto> map(List<Message> messages){
+        return messages.stream()
+                .map(this::map)
+                .collect(Collectors.toList());
+    }
+
+    public MessageDto map(Message message){
+
+        User receiverUser = userRepository
```

Comment:
If you use it with list, it's veeery inefficent, because there will be N*2 queries to db for getting this users.
I'd create another private method `map`, that accepts `message`, `receiverUser `, `senderUser ` and performs mapping.
This method then will call it with two users, that it gets from db.
Method for list will call it with two collections of users, obtained by just two queries execution.